test/Twig/Tests/Node/Expression/GetAttrTest.php

```diff
@@ -37,7 +37,7 @@ public function getTests()
         $tests[] = array($node, sprintf('%s%s, "bar", array())', $this->getAttributeGetter(), $this->getVariableGetter('foo', 1)));
 
         $node = new Twig_Node_Expression_GetAttr($expr, $attr, $args, Twig_Template::ARRAY_CALL, 1);
-        $tests[] = array($node, sprintf('%s%s, "bar", array(), "array")', $this->getAttributeGetter(), $this->getVariableGetter('foo', 1)));
+        $tests[] = array($node, sprintf('(is_array(%s) || ($context["foo"] ?? null) instanceof ArrayAccess ? (($context["foo"] ?? null)["bar"] ?? null) : null)', $this->getVariableGetter('foo', 1), $this->getVariableGetter('foo', 1)));
```

Comment:
shouldn't `($context["foo"] ?? null)` use `$this->getVariableGetter` here too ?

Reply:
These abstractions were useful in 1.x, not so much now. I've kept them to ease merging, but no need to keep them for new code.